boruta/1-click-boruta/script.whizzml

```diff
@@ -0,0 +1,386 @@
+  (let (to-names (lambda (x) (get-in fields [x "name"] false))
+        important (map to-names
+                       (keys (filter-map (lambda(x y) (> y 0))
+                                         f-rank)))
+        important (append important objective)
+        undecided (map to-names
+                       (keys (filter-map (lambda(x y) (= y 0))
+                                         f-rank)))
+        unimportant (filter (lambda (x) (not (or (member? x important)
+                                                 (member? x undecided))))
+                            f-names))
+    {"important" important
+     "unimportant" unimportant
+     "undecided" undecided}))
+
+;; filter-map
+;; Filters a map by applying the corresponding filter function to its keys
+;;
+;; Inputs:
+;;   fn: (function) function to be applied to the map keys and/or values
+;;   map: (map) Map to be filtered
+;;
+;; Output: (map) Filtered map
+(define (filter-map fn map)
+    (iterate (out-map {} key (keys map))
+      (if (fn key (get map key false))
+          (assoc out-map key (get map key false))
+          out-map)))
+
+;; shadow-ds
+;; Extends a dataset adding a shadow field for each field used in input_fields.
+;; If the number of shadow fields is less than 5, fields are repeated.
+;;
+;; Inputs
+;;  ds-id: (dataset-id) Dataset ID
+;;  f-ids: (list) List of field IDs that will be used as input and extended
+;;                with shadow fields
+;;
+;; Output: (list) List of elements [dataset-id shadow-field-ids fields]
+;;   dataset-id: (dataset-id) ID of the dataset with shadow fields
+;;   shadow-fields-ids: (list) List of IDs for the newly added shadow fields
+;;   fields: (map) New fields structure for the fields in the original dataset.
+;;                 This structure is needed to know the IDs in the
+;;                 shadow dataset for the original fields
+;;
+(define (shadow-ds ds-id fields obj-name)
+  (let (f-id-names (reduce (lambda (x y)
+                             (assoc x y (get-in fields [y "name"] false)))
+                           {}
+                           (keys fields))
+        obj-f (filter-map (lambda (x _)
+                            (= (get-in fields [x "name"] false) obj-name))
+                          f-id-names)
+        obj-id (head (keys obj-f))
+        f-id-names (filter-map (lambda (x _) (!= x obj-id)) f-id-names)
+        f-ids (keys f-id-names)
+        input-fields (cons obj-id f-ids)
+        f-ids (repeat-fields f-ids 5)
+        f-names (map (lambda (x) (str "shadow " (get f-id-names x false)))
+                     f-ids)
+        flatline-fs (map (lambda (x) [(flatline
+                                        "(weighted-random-value {{x}})")
+                                      (str "shadow " (get f-id-names x))])
+                         f-ids)
+        new-fields (lambda (x) (assoc {} "field" (head x)
+                                         "name" (head (tail x))))
```

Comment:
or, since you have `f-names` you can also use a map, again without `flatline-fs`:

```
new-fields (map (lambda (id name) {"field" id "name" name})
                f-ids
                f-names)
```